scripts/upload_xgboost_mnist_dataset/upload_xgboost_mnist_dataset

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+import argparse
+import boto3
+import csv
+import gzip
+import io
+import numpy as np
+import pickle
+import sys
+import urllib.request
+
+DATA_URL = 'http://deeplearning.net/data/mnist/mnist.pkl.gz'
+
+def main():
+
+    s3_bucket, s3_prefix = parse_args()
+
+    for data_partition_name, csv_data in convert_data_to_csv(get_data_partitions(DATA_URL)):
+        key = f'{s3_prefix}/{data_partition_name}/examples'
+        print(f'Uploading {csv_data.getbuffer().nbytes} bytes to s3://{s3_bucket}/{key}')
+        boto3.Session().resource('s3').Bucket(s3_bucket).Object(key).upload_fileobj(csv_data)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Helper script that splits the MNIST dataset into train, validation, and test sets, then uploads them in CSV format to the specified s3 bucket')
+    parser.add_argument('--s3-bucket', type=str, required=True)
+    parser.add_argument('--s3-prefix', type=str, required=True)
```

Comment:
Can we provide a solid default for this argument? Perhaps the root prefix `/`?

Reply:
I don't really like this because the script creates three directories; if we force user to provide prefix it helps them better organize their data.